### PR TITLE
feat(website): add index sections to reference and resources

### PR DESCRIPTION
**Pull request by developerkunal**
This commit adds new sections to the File Description Page for Taiko Protocol resources, including guidelines for contributing to Taiko Protocol on GitHub, official links, presentations, a Taiko glossary, and the official whitepaper. These additions should make it easier for developers and users to navigate and access relevant information about Taiko Protocol.

@jscriptcoder

### CODE REVIEW

**Comment by jscriptcoder** [comment]
Hi @developerkunal , thanks a lot for your contribution. I think @d1onys1us would be the right person to review this one.

**Comment by d1onys1us** [comment]
@developerkunal i don't mean to block this PR, we could very well merge it in and it would be helpful for users, but there is one issue with this approach which is maintainability.

as we add more documentation to these sections the documentation maintainers will always need to remember to update these sections. and also the names for the sections such as "resources" "reference" "contract documentation" should be semantic enough to describe what is under that section without needing a description.

but i agree this is really useful as an index / table of contents. the larger initiative / correct approach would actually be if we could write a script somehow that will parse all of the files automatically, and automatically generate an index / table of contents dynamically. this is provided out of the box by docusaurus, but not nextra (i don't think).

let me know if this issue sounds interesting and if you want to work on doing it with this approach, otherwise i can review this PR and i think we can merge it in and accept the slightly larger maintenance overhead. thanks for your contribution ❤️

**Comment by developerkunal** [comment]
@d1onys1us thanks for reply i would love to work on create a script to add table of content but at the moment it will not be that fast if we can merge this and when i will be ready to update script we can remove this content.the issue was why i created it like this because it contains a bit of description which is good for SEO and Google Indexing But i surely can work on the new issue.

File: packages/website/pages/docs/reference.mdx
Comment (d1onys1us) [inline packages/website/pages/docs/reference.mdx:19]: can you move this to the top under "handy reference documentation for Taiko"

File: packages/website/pages/docs/resources.mdx
Comment (d1onys1us) [inline packages/website/pages/docs/resources.mdx:23]: same here

File: packages/website/pages/docs/reference.mdx
Comment (d1onys1us) [inline packages/website/pages/docs/reference.mdx:13]: can you please remove the sentences like "this is essential / crucial for building applications on the blockchain". in contract addresses, protocol constants, and rpc configuration.

because of docs philosophy here: https://github.com/taikoxyz/taiko-mono/blob/main/CONTRIBUTING.md#philosophy (minimum viable documentation)

File: packages/website/pages/docs/resources.mdx
Comment (d1onys1us) [inline packages/website/pages/docs/resources.mdx:21]:
```suggestion
This section contains the official Taiko Protocol whitepaper. It includes a detailed overview of the protocol, its benefits, and its use cases.
```

**Comment by d1onys1us** [comment]
@developerkunal understood, we can bring this in for the time being. but if it starts to get stale or become a maintenance overhead we should remove it. but ideally, we can create that script and work with nextra to get the feature in the right way. thanks :)

**Comment by developerkunal** [comment]
@d1onys1us i have maded the fixes

**Review by dionysuzx — APPROVED** [review]

**Review by dantaik — APPROVED** [review]

**Pull request merged**
